code/game/objects/items/weapons/flamethrower.dm

```diff
@@ -218,3 +218,11 @@
 	..()
 	ptank = new /obj/item/weapon/tank/internals/plasma/full(src)
 	update_icon()
+
+
+/obj/item/weapon/flamethrower/hit_reaction(mob/living/carbon/human/owner, attack_text, damage, attack_type)
+	if(damage && attack_type == PROJECTILE_ATTACK && prob(5) && ptank)

```

Comment:
```ptank``` should be first, so that flamethrowers without tanks fail the check instantly, rather than in the 4th condition.